### PR TITLE
feat(security): Allow to opt-out of ratelimit protection, e.g. for te…

### PR DESCRIPTION
**Pull request by nickvergessen**
…sting on CI

* Resolves: # 

## Summary

On talk integration tests are red on CI since middle of last week (Merge of https://github.com/nextcloud/server/pull/37497 ). Turns out it's the rate limiting hitting in disallowing to set the user's display name. See https://github.com/nextcloud/spreed/pull/9233 for debugging history.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

File: lib/private/Security/RateLimiting/Backend/DatabaseBackend.php [hunk 1]
```diff
@@ -115,7 +118,12 @@ public function registerAttempt(string $methodIdentifier,
 			->values([
 				'hash' => $qb->createNamedParameter($identifier, IQueryBuilder::PARAM_STR),
 				'delete_after' => $qb->createNamedParameter($deleteAfter, IQueryBuilder::PARAM_DATE),
-			])
-			->executeStatement();
+			]);
+
+		if (!$this->config->getSystemValueBool('ratelimit.protection.enabled', true)) {
```
Comment (juliushaertl) [inline lib/private/Security/RateLimiting/Backend/DatabaseBackend.php:123]: Any reason to not return early at the beginning of the method? Would feel easier to read to me.

**Review by juliusknorr — APPROVED** [review]

File: lib/private/Security/RateLimiting/Backend/DatabaseBackend.php (hunk 1 above)
Comment (nickvergessen) [reply lib/private/Security/RateLimiting/Backend/DatabaseBackend.php:123]: I'm drown apart between:
* keep code as simple as possible (return early)
* Keep code as close to production as possible as soon every dev will have this active locally

**Pull request merged**